src/gateway/sharding/shard_runner.rs

```diff
@@ -487,14 +467,29 @@ impl ShardRunner {
     }
 
     #[cfg_attr(feature = "tracing_instrument", instrument(skip(self)))]
-    async fn update_manager(&self) {
-        self.manager
-            .update_shard_latency_and_stage(
-                self.shard.shard_info().id,
-                self.shard.latency(),
-                self.shard.stage(),
-            )
-            .await;
+    async fn update_runner_info(&self) {
+        let mut runner_info = self.runner_info.lock().await;
```

Comment:
Can you convert this to `try_lock` and just ignore if it can't grab the lock? This makes me worried that user code could hold the runner_info lock too long and just lead to a shard dying.

Thinking about it, the runners lock could also be changed to a sync lock as it's never needed to be held over an await sensibly.

Reply:
Changed it - can you think of a circumstance where one of the locks might end up getting held across an await point? Just so I have an example in mind

Reply:
A user could loop over the runners reporting statuses and forget to drop the lock before sending the messages.

Reply:
Do you think this is likely enough that the Mutexes should remain async?

Reply:
Nope, that's why I said to make them sync lol